shell/browser/electron_permission_manager.cc

```diff
@@ -210,6 +210,7 @@ void ElectronPermissionManager::RequestPermissionsWithDetails(
   int request_id = pending_requests_.Add(std::make_unique<PendingRequest>(
       render_frame_host, permissions, std::move(response_callback)));
 
+  details.Set("requestingOrigin", request_description.requesting_origin.spec());
```

Comment:
`WebContentsPermissionHelper` provides the WebContents last commited URL
https://github.com/electron/electron/blob/93a6f3e60739a58dedb5b50563e0cb8e7ed46e29/shell/browser/web_contents_permission_helper.cc#L215-L218

`FileSystemAccessPermissionContext` provides the RenderFrameHost's last committed origin
https://github.com/electron/electron/blob/93a6f3e60739a58dedb5b50563e0cb8e7ed46e29/shell/browser/file_system_access/file_system_access_permission_context.cc#L298-L325

We should try to resolve this conflict before exposing this property. Otherwise, it might be confusing if a child frame requested a permission and `requestingOrigin` didn't match the child frame.

Alternatively, we can expose the RFH itself so that `details.frame.origin` can be accessed directly. This would reduce the need for additional documentation as well.

```suggestion
  details.SetGetter("frame", render_frame_host);
  details.Set("requestingOrigin", request_description.requesting_origin.spec());
```